js/web/lib/wasm/proxy-messages.ts

```diff
@@ -12,51 +15,61 @@ export type GpuBufferMetadata = {
   dispose?: () => void;
 };
 
+/**
+ * Tensors on location 'cpu-pinned' and 'gpu-buffer' are not serializable.
+ */
 export type UnserializableTensorMetadata =
     [dataType: Tensor.Type, dims: readonly number[], data: GpuBufferMetadata, location: 'gpu-buffer']|
     [dataType: Tensor.Type, dims: readonly number[], data: Tensor.DataType, location: 'cpu-pinned'];
 
+/**
+ * Tensor metadata is a tuple of [dataType, dims, data, location], where
+ * - dataType: tensor data type
+ * - dims: tensor dimensions
+ * - data: tensor data, which can be one of the following depending on the location:
+ *   - cpu: Uint8Array
+ *   - cpu-pinned: Uint8Array
+ *   - gpu-buffer: GpuBufferMetadata
+ * - location: tensor data location
+ */
 export type TensorMetadata = SerializableTensorMetadata|UnserializableTensorMetadata;
 
 export type SerializableSessionMetadata = [sessionHandle: number, inputNames: string[], outputNames: string[]];
 
-export type SerializableModeldata = [modelDataOffset: number, modelDataLength: number];
+export type SerializableInternalBuffer = [bufferOffset: number, bufferLength: number];
```

Comment:
nit: How do you think if we continue keep Model in the name, like SerializableInternalModelBuffer? And `MessageCopyFromExternalBuffer` -> `MessageCopyFromExternalModelBuffer` so that we can know the interalBuffer or externalBuffer both mean the model data. Anyway, it doesn't impact you merge this PR:).

Reply:
I explicitly change the name because for the data itself, it is just representing a sub-view of the memory buffer and it's not necessarily to be model data. This is because I think "copyFromExternalBuffer" is better name than "createSessionAllocate" for what the function actually does.

Reply:
Got it, thanks.